notebooks/visit_data.py

```diff
@@ -0,0 +1,255 @@
+# can we predict the next site to be visited from the current site?
+# this is a hard high cardinality challenge, so at random, the predictive performance is low
+# is this flawed anyway, as the results being presented just reflect what the user was limited to selecting
+# like predicting that the bus is red
+# have a performance measurement at random, and using the model results
+# are the parent pagePaths sufficiently granular
+
+import pandas as pd
+import os
+import numpy as np
+import re
+from flatten_dict import unflatten, flatten
+
+os.chdir("/home/james/Downloads")
+data_dir = '/home/james/Documents/gds_nlp/govuk-entity-personalisation/data'
+
+# downloaded data from bigquery, using sql beneath
+'''
+SELECT
+date, fullVisitorId, visitId, hits.hour, hits.minute, hits.hitNumber, hits.page.pagePath, hits.page.pageTitle
+FROM `govuk-bigquery-analytics.87773428.ga_sessions_20201113`, unnest(hits) as hits
+WHERE hits.type = "PAGE" and hits.hour = 12
+Limit 10000000
+'''
+visit_data = pd.read_csv("bq-results-20201116-153658-4lgqv331oo43.csv")
```

Comment:
Following on with making it more generalisable through utilising relative paths, can tweak this to:
```suggestion
visit_data = pd.read_csv(DATA_DIR + "bq-results-20201116-153658-4lgqv331oo43.csv")
```